.github/workflows/check.yml

```diff
@@ -0,0 +1,65 @@
+name: Combined Checks
+
+# Controls when the action will run. Triggers the workflow on push or pull request
+# events but only for the main branch
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  combined-checks:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ['3.8', '3.9', '3.10', '3.11', '3.12']
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+
+    - name: Install dependencies for mypy
+      if: matrix.python-version != '3.8'
+      run: pip install mypy
+
+    - name: Configure mypy globally
+      if: matrix.python-version != '3.8'
+      run: echo -e "[mypy]\nignore_missing_imports = True" > mypy.ini
+
+    - name: Run mypy
+      if: matrix.python-version != '3.8'
+      run: mypy .
+
+    - name: Cache poetry install
+      uses: actions/cache@v2
+      with:
+        path: ~/.local
+        key: poetry-1.7.1-py${{ matrix.python-version }}-20231217
+
+    - uses: snok/install-poetry@v1
+      with:
+        version: 1.7.1
+        virtualenvs-create: true
+        virtualenvs-in-project: true
+
+    - name: Install dependencies with Poetry
+      run: poetry install --no-interaction
```

Comment:
@ctheune I think this may be related to the issue in #108 previously we were calling --no-root which makes sense why tests were failing because it wasn't finding pycountry as we were telling poetry to not install it